src/sortStudents.ts

```diff
@@ -1,16 +1,60 @@
+export enum SortType {
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
+}
+
+export type SortOrder = 'asc' | 'desc';
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
-export enum SortType {
-  // describe SortType enum
-}
+function getAverageGrade(grades: number[]): number {
+  const sum = grades.reduce((total, grade) => total + grade, 0);
 
-// create SortOrder type
-export type SortOrder;
+  return sum / grades.length;
+}
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortedStudents = [...students];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  switch (sortBy) {
+    case SortType.Name:
+      return order === 'asc'
+        ? sortedStudents.sort((a, b) => a.name.localeCompare(b.name))
+        : sortedStudents.sort((a, b) => b.name.localeCompare(a.name));
+    case SortType.Surname:
+      return order === 'asc'
+        ? sortedStudents.sort((a, b) => a.surname.localeCompare(b.surname))
+        : sortedStudents.sort((a, b) => b.surname.localeCompare(a.surname));
```

Comment:
Checklist: `[CODE STYLE] - DRY. Combine logic for sorting by fields with same data types.`

```suggestion
  const sortOrder: number = order === 'asc' ? 1 : -1;
  switch (sortBy) {
    case SortType.Name:
    case SortType.Surname:
      return sortedStudents.sort((a, b) => sortorder * a[sortBy].localeCompare(b[sortBy]));
```

Do the same for Age and Married